op-program/host/config/config.go

```diff
@@ -162,7 +170,18 @@ func NewConfigFromCLI(log log.Logger, ctx *cli.Context) (*Config, error) {
 	if l2Head == (common.Hash{}) {
 		return nil, ErrInvalidL2Head
 	}
-	l2OutputRoot := common.HexToHash(ctx.String(flags.L2OutputRoot.Name))
+	var l2OutputRoot common.Hash
+	var agreedPrestate []byte
+	if ctx.IsSet(flags.L2OutputRoot.Name) {
```

Comment:
Note that if both `L2OutputRoot` and `L2AgreedPrestate` flags are set, then the former overrides the latter with `agreedPrestate` remaining nil. Ensuring both flags are mutually exclusive fixes this.

Reply:
Flags can't both be specified and we now verify consistency in the config checks to cover programmatic users.